### DISCUSSION TITLE
High-genus surface generation

### DISCUSSION THREAD

**Discussion by aaron-jesse-ol** [weria-pezeshkian/FreeDTS#2] [Q&A] [answered]
I'm curious how to reproduce the high-genus examples shown in [the preprint](https://www.biorxiv.org/content/10.1101/2023.05.05.539540v1); the GEN option high_gen appears to be undocumented. Any tips?

Secondly, can GEN make high-genus surfaces with doubly or triply PBCs?

**Comment by weria-pezeshkian** [comment, marked as answer]
Any closed surface with or without PBC can be handled by FreeDTS. As how to make them, it depends. GEN can only create  closed high genus without PBC at the moment. GEN was intend only for simple surfaces to start. I will soon add the high-gen description to the GEN script.  Just use -type high_gen and -g genus.

**Reply by aaron-jesse-ol** [reply]
I will be curious to see that description of high_gen usage when you have it, because my naïve attempts have been unsuccessful. For example, trying -g 2 I get "Error: projection is zero error":

```
% ../GEN -box 50 50 50 -type high_gen -g 2
----------------- GEN ---------------- 
------ Generate TS file  ------------ 
---------------------------- Weria Pezeshkian -------------------------------------- 
 number of the vertex 50 total number of triangles 104

% cat input.dts
Integrator                      = MC
MC_Moves                        = 1 1 1
Initial_Step                    = 1
Box_Centering_F                 = 0
Final_Step                      = 10000
Display_periodic                = 1000
OutPutEnergy_periodic           = 100
Restart_periodic                = 10000
Kappa                           = 10 0
OutPutTRJ_TSI                   = 1000  10  TrjTSI
GeneralOutputFilename           = output
Cell_Size                       = 2.5 2.6 2.5

% cat top.top
out.q 10

% ../DTS -in input.dts -top top.top -seed 76532
----------------- DTS ---------------- 
------ Simulation of Biomembranes Using Dynamically Triangulated Surfaces ------------ 
---------------------------- Weria Pezeshkian -------------------------------------- 
warning: no inclsuion interaction energy is mentioned in the inputfile, they are all set to zero
trinagle is read 
-----> Note: Mesh was taken from the topology  
output file name and temp 1.00000000
Links normall vector error 
Links normall vector error 
Links normall vector error 
Links normall vector error 
Links normall vector error 
Links normall vector error 
Links normall vector error 
Links normall vector error 
-----> Error: projection is zero error
```

It's unclear to me what's causing this. This is the output from GEN:
<img width="677" alt="Screenshot 2023-05-10 at 16 36 28" src="https://github.com/weria-pezeshkian/FreeDTS/assets/1960432/bfd1c9ba-5183-4d97-b465-ad4ddd697eb3">

**Comment by weria-pezeshkian** [comment]
I think it is just because you system is small. add -N 8 to your command line

**Reply by aaron-jesse-ol** [reply]
Ah! That was indeed the problem, thanks. I'd noticed that having N < sqrt(g) threw a warning and assumed that any larger N would be sufficient, but clearly that is just a rough guideline.
